src/components/Upload.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import UploadForm from './UploadForm';
+import { withStyles } from '@material-ui/core/styles';
+
+const progressStyle = {
+  color: 'black',
+};
+
+const styles = {
+  success: {
+    fontWeight: 'bold',
+    marginTop: 10,
+    color: 'green',
+  },
+  error: {
+    fontWeight: 'bold',
+    marginTop: 10,
+  },
+};
+
+class Upload extends Component {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      file: '{\n' +
+      '  "entry": [\n' +
+      '    {\n' +
+      '      "fullUrl": "urn:uuid:5f0ed72c-1d71-4f87-bf44-a23a7e506222",\n' +
+      '      "resource": {\n' +
+      '        "resourceType": "Basic",\n' +
+      '        "id": "5f0ed72c-1d71-4f87-bf44-a23a7e506222",\n' +
+      '        "meta": {\n' +
+      '          "profile": [\n' +
+      '            "http://standardhealthrecord.org/fhir/StructureDefinition/shr-entity-Person"\n' +
+      '          ]\n' +
+      '        },\n' +
+      '        "code": {\n' +
+      '          "coding": [\n' +
+      '            {\n' +
+      '              "system": "http://standardhealthrecord.org/fhir/basic-resource-type",\n' +
+      '              "code": "shr-entity-Person",\n' +
+      '              "display": "shr-entity-Person"\n' +
+      '            }\n' +
+      '          ]\n' +
+      '        }\n' +
+      '      }\n' +
+      '    }\n' +
+      '  ]\n' +
+      '}',
+      public_key: '',
+      metadata: '',
+    };
+  }
+
+  onInputChange = (event) => {
+    const property = event.target.id;
+    const value = event.target.value;
+    this.setState({ [property]: value });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const file = JSON.parse(this.state.file);
```

Comment:
Dispatch an upload error here if file is un-parsable? Most people (myself included) suck at hand rolling json so I imagine this will fail a lot

Reply:
Done, moved validation further downstream to the action.